### PR TITLE
Alerts | Amenity | Escalators/Elevators | # of working Elevators & Escalators

### PR DESCRIPTION
**Pull request by i0sea**
#### Summary of changes

**Asana Ticket:** [Alerts | Amenity | Escalators/Elevators | # of working Elevators & Escalators](https://app.asana.com/0/385363666817452/1204305469426567/f)

[Amenity Modal | Escalators](https://app.asana.com/0/385363666817452/1204401358635410/f)

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

File: apps/site/assets/ts/stop/components/StationInformation.tsx [hunk 1]
```diff
@@ -32,6 +36,9 @@ const StationInformation = ({
     }
   });
 
+  const elevators = facilitiesByType.get("ELEVATOR")!;
```
Comment (thecristen) [inline apps/site/assets/ts/stop/components/StationInformation.tsx:39]: **suggestion:** use the fallback `[]` here, e.g. change this to `facilitiesByType.get("ELEVATOR") || []`, which then simplifies passing this variable into `<ElevatorsAmenityCard>`. 

(same with escalators)

File: apps/site/assets/ts/stop/components/amenities/access-amenities-helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Alert, Facility } from "../../../__v3api";
+import Badge from "../../../components/Badge";
+
+export const availabilityMessage = (
+  brokenElevators: number,
+  totalElevators: number,
+  facilityType: string
```
Comment (thecristen) [inline apps/site/assets/ts/stop/components/amenities/access-amenities-helpers.tsx:8]: question: should this be typed "Escalators" | "Elevators" or some such?

File: apps/site/assets/ts/stop/components/amenities/access-amenities-helpers.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Alert, Facility } from "../../../__v3api";
+import Badge from "../../../components/Badge";
+
+export const availabilityMessage = (
+  brokenElevators: number,
+  totalElevators: number,
```
Comment (thecristen) [inline apps/site/assets/ts/stop/components/amenities/access-amenities-helpers.tsx:7]: question: this is used for escalators too right? I wonder if a more generic name here might be readable, maybe `facilityAlertsLength` and `facilityLength`

File: apps/site/assets/ts/models/alert.ts [hunk 4]
```diff
@@ -231,3 +231,16 @@ export const hasFacilityAlert = (
     return false;
   });
 };
+
+export const alertsByFacility = (
```
Comment (thecristen) [inline apps/site/assets/ts/models/alert.ts:235]: nice! can you add a test for this?

File: apps/site/assets/ts/stop/__tests__/components/amenities/Access-Amenities-HelpersTest.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+const alert1: Alert = {
+  severity: 7,
+  priority: "high",
+  lifecycle: "new",
+  informed_entity: { facility: "703" } as InformedEntity,
+  active_period: [["2020-09-10 08:00", "2020-09-10 20:00"]]
+} as Alert;
+
+describe("availabilityMessage", () => {
+  it("should renturn 'View available facilityType' if at least one is available", () => {
```
Comment (thecristen) [inline apps/site/assets/ts/stop/__tests__/components/amenities/Access-Amenities-HelpersTest.tsx:34]: typo: should return

**Review by thecristen — CHANGES_REQUESTED** [review]
🛗  I do wonder if consolidating the shared logic between `<ElevatorsAmentityCard />` and `<EscalatorsAmenityCard />` into a shared `<AccessAmenitiesCard />` component (containing the functionality from the `access-amenities-helpers` file) might make things a bit more organized, but this is overall pretty nice! 
Requesting changes to add a test for the new function in `alert.ts`.

**Comment by i0sea** [comment]
@thecristen Considered creating an access amenity card but that seemed less straightforward than just using the helper class in both elevator and escalator amenity cards. Also makes it easier to make changes to each if such a case arises in the future. Made all other changes

**Review by thecristen — APPROVED** [review]
LGTM

**Pull request merged**